### DISCUSSION TITLE
Error coremark benchmark as workload on gem5

### DISCUSSION THREAD

**Discussion by waseemabbas-10xe** [gem5/gem5#1132] [gem5-dev]
![image](https://github.com/gem5/gem5/assets/142871238/e1ce58ca-457c-4b58-a46c-af527ae6d1bc)
command i run :./build/RISCV/gem5.opt ./configs/example/se.py -c tests/test-progs/hello/bin/riscv/linux/coremark.bare.riscv

**Comment by robhau** [comment]
Is `coremark.bare.riscv` a program written by you? I only have `hello` in `tests/test-progs/hello/bin/riscv/linux/`.

It seems that you try to access the `mstatus` register in usermode. However, `mstatus` is only accessible in machine mode. I suggest you to take a look into the RISC-V specification (privileged architecture).

I have no experience with SE mode, since I do only full system simulation. However, I guess that in SE mode, you are only in the user mode, since the OS is emulated by your host computer. Therefore, you will never be able to access `mstatus` in SE.

**Comment by waseemabbas-10xe** [comment]
Yes, You are saying right @robhau I have find this is only way that if I run gem5 in Full system mode then I can access the mstatus in the machine mode. @robhau  can you give me full_system python script and which command you are running.

**Reply by robhau** [reply]
You can find a minimal working bare metal example here: https://github.com/robhau/gem5/blob/minor_cpu_bug_minimal_working_example/configs/control_hazard.py (It is very simple because of its purpose)

**Comment by waseemabbas-10xe** [comment]
No I have downloaded it from the git repo and then i generate the hex file
using the toolchain and then run ingem5 as a workload.

**Reply by robhau** [reply]
I suggest you [this tutorial](https://five-embeddev.com/baremetal/intro/) for RISC-V bare metal programming.

**Comment by waseemabbas-10xe** [comment]
Thanks a lot. I am going to see this tutorial and let you know.Thank you
